### PR TITLE
Added 'C++ standard' build option for Windows / VS2022 support

### PR DESCRIPTION
**Pull request by azurepine**

### CODE REVIEW

File: build/windows/build_windows.py [hunk 1]
```diff
@@ -29,14 +29,16 @@
...
+parser.add_argument('-t', '--tool', help='platform toolset: v140, v141, v142, v143...')
+parser.add_argument('-s', '--stdcpp', help='C++ standard: 11, 14, 17, 20...')
 parser.add_argument(      '--dll',  help='build DLL', action='store_true')
 
 args = parser.parse_args()
 
 BUILD_MODE = 'Debug'
-ARCH = 'x64'
-TOOLSET = args.tool
+ARCH = 'x86'
```
Comment (Dimon4eg) [inline build/windows/build_windows.py:39]: I think by default should x64 as it's more popular

File: build/windows/build_windows.py [hunk 2]
```diff
@@ -124,6 +144,9 @@ def copy_dll(dest):
   '-DCMAKE_SYSTEM_VERSION=' + SYSTEM_VERSION
 ]
 
+if CXX_STANDARD != '':
+    cmake_cmd.extend(['-DCMAKE_CXX_STANDARD=' + CXX_STANDARD])
+
```
Comment (Dimon4eg) [inline build/windows/build_windows.py:149]: better move this to build_common.py to support this on all platforms

Comment (azurepine) [reply build/windows/build_windows.py]: https://github.com/heroiclabs/nakama-cpp/pull/52/commits/434d691af72804bd015d3887d1ed9734c3ee7b50

File: build/windows/build_windows.py (hunk 1 above)
Comment (azurepine) [reply build/windows/build_windows.py]: https://github.com/heroiclabs/nakama-cpp/pull/52/commits/501c729d50332d81a72eb7e323c0c35412d96815

**Comment by lugehorsam** [comment]
We've rewritten our build system within which there is VS 2022 support.

**Pull request closed without merging**